implementations/rust/ockam/ockam_command/src/project/enroll.rs

```diff
@@ -191,6 +183,89 @@ impl EnrollCommand {
 
         Ok(project)
     }
+
+    fn plain_output(
+        &self,
+        identity_name: &str,
+        project_name: &str,
+        credential_data: &CredentialData,
+    ) -> Result<String> {
+        let mut buf = String::new();
+        writeln!(
+            buf,
+            "{}",
+            fmt_ok!(
+                "Successfully enrolled identity {} to the {} project.\n",
+                color_primary(identity_name),
+                color_primary(project_name)
+            )
+        )?;
+
+        writeln!(
+            buf,
+            "{}",
+            fmt_log!("The identity has a credential in this project")
+        )?;
+        writeln!(
+            buf,
+            "{}",
+            fmt_log!(
+                "created at {} that expires at {}\n",
+                color_primary(human_readable_time(credential_data.created_at)),
+                color_primary(human_readable_time(credential_data.expires_at))
+            )
+        )?;
+
+        if !credential_data.subject_attributes.map.is_empty() {
+            writeln!(
+                buf,
+                "{}",
+                fmt_log!(
+                    "The following attributes are attested by the project's membership authority:"
+                )
+            )?;
+            for (k, v) in credential_data.subject_attributes.map.iter() {
+                let k = std::str::from_utf8(k).unwrap_or("**binary**");
+                let v = std::str::from_utf8(v).unwrap_or("**binary**");
+                writeln!(
+                    buf,
+                    "{}",
+                    fmt_log!(
+                        "{}{}",
+                        fmt::INDENTATION,
+                        color_primary(format!("\"{k}={v}\""))
+                    )
+                )?;
+            }
+        }
+        Ok(buf)
+    }
+
+    fn extract_credential_data(
```

Comment:
This could be moved to the `CredentialAndPurposeKey` struct in order to be reused in other places.

Reply:
We actually had a method to extract just the needed data (credential data only without the purpose key data). I've removed the unnecessary code.